### PR TITLE
Added Monte Carlo in Swift 4.1

### PR DESCRIPTION
**Pull request by CDsigma**
Added Monte Carlo in Swift 4.1

### CODE REVIEW

File: chapters/monte_carlo/monte_carlo.md [hunk 1]
```diff
@@ -113,6 +115,8 @@ Feel free to submit your version via pull request, and thanks for reading!
 {% sample lang="java" %}
 ### Java
 [import, lang:"java"](code/java/MonteCarlo.java)
+### Swift
```
Comment (jiegillet) [inline chapters/monte_carlo/monte_carlo.md]: You're missing `{% sample lang="swift" %}` here

File: chapters/monte_carlo/code/swift/monte_carlo.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    public static func random(min: Double, max: Double) -> Double {
+
+        return Double.random * (max - min) + min
+    }
+}
+
+
+func isInCircle(x: Double, y: Double, radius: Double) -> Bool {
+
+    return (x*x) + (y*y) < Double(radius*radius)
```
Comment (jiegillet) [inline chapters/monte_carlo/code/swift/monte_carlo.swift]: Do you need to cast ` Double(radius*radius)` even though you have `radius: Double `?

File: chapters/monte_carlo/monte_carlo.md (hunk 1 above)
Comment (CDsigma) [reply chapters/monte_carlo/monte_carlo.md]: Sorry about that just committed a change to fix it

File: chapters/monte_carlo/code/swift/monte_carlo.swift (hunk 2 above)
Comment (CDsigma) [reply chapters/monte_carlo/code/swift/monte_carlo.swift]: You're totally right I just committed a change to remove that redundancy

**Comment by jiegillet** [comment]
Sorry for the back and forth, I had written a review but had not realized it hadn't been shared yet!

File: chapters/monte_carlo/code/swift/monte_carlo.swift [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        randX = Double.random(min: 0, max: radius)
+        randY = Double.random(min: 0, max: radius)
+
+        if(isInCircle(x: randX, y: randY, radius: radius)) {
+            piCount += 1
+        }
+    }
+    
+    let piEstimate = Double(4 * piCount)/(Double(n))
+    print("Percent error is: \(100*(Double.pi - piEstimate)/Double.pi)%")
```
[the remark this answers is not in the document]
Comment (CDsigma) [reply chapters/monte_carlo/code/swift/monte_carlo.swift]: Good suggestion placing it in main() makes more sense. I just made a commit which addresses this.

File: chapters/monte_carlo/code/swift/monte_carlo.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import Foundation
+
+
+
+public extension Double {
+
+    public static var random: Double {
+        
+        return Double(arc4random()) / 0xFFFFFFFF
```
[the remark this answers is not in the document]
Comment (CDsigma) [reply chapters/monte_carlo/code/swift/monte_carlo.swift]: Sounds good added a comment which says what that line of code will do

File: chapters/monte_carlo/code/swift/monte_carlo.swift [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    
+    let piEstimate = Double(4 * piCount)/(Double(n))
+    print("Percent error is: \(100*(Double.pi - piEstimate)/Double.pi)%")
+
+    return piEstimate
+}
+
+
+func main() {
+    print("Pi estimate is: ", monteCarlo(n: 10000, radius: 50))
```
[the remark this answers is not in the document]
Comment (CDsigma) [reply chapters/monte_carlo/code/swift/monte_carlo.swift]: Sounds good I just updated my code accordingly 👍

**Comment by jiegillet** [comment]
Thank you, looks great now :)

**Pull request merged**